### PR TITLE
Provide how to guide for verifying signed container images

### PR DESCRIPTION
**Pull request by dimitar-dimitrow**
[#307] Provide how to guide for verifying signed container images

Signed-off-by: Dimitar Dimitrov <dimitar.dimitrov3@bosch.com>

### CODE REVIEW

File: web/site/content/docs/how-to-guides/verify-signed-container-images.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+---
+title: "Verify signed container images"
+type: docs
+description: >
+    Verify that container image is signed when creating a container from it in Kanto Container Management.
+weight: 5
+---
+
+By following the steps below you will sign a container image and push it to a local registry using a{{% refn "https://github.com/notaryproject/notation" %}}`notation`{{% /refn %}}. Then notation trust policy and Kanto Container Management be configured in a way that running containers from the signed image via kanto-cm CLI will be successfull, while running containers from unsigned images will fail.
```
Comment (k-gostev) [inline web/site/content/docs/how-to-guides/verify-signed-container-images.md:9]:
```suggestion
By following the steps below, you will sign a container image and push it to a local registry using a{{% refn "https://github.com/notaryproject/notation" %}}`notation`{{% /refn %}}. Then a notation trust policy and the Kanto Container Management service will be configured in a way that running containers from the signed image via kanto-cm CLI will be successful, while running containers from unsigned images will fail. 
```

File: web/site/content/docs/how-to-guides/verify-signed-container-images.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+{{% tip %}}
+When signing and verifying container images it is recommended to use the image digest instead of a tag as the digest is immutable.
+{{% /tip %}}
+Get the image digest and assign it to an environment variable to be used the next steps of the guide:
+```shell
+export IMAGE=$(sudo docker inspect --format='{{index .RepoDigests 0}}' localhost:5000/dummy-hello:signed)
+echo $IMAGE
+```
+
+Generate a key-pair with notation and add it as the default key to be used by notation when signing:
```
Comment (k-gostev) [inline web/site/content/docs/how-to-guides/verify-signed-container-images.md:44]:
```suggestion
Generate a key-pair with notation and add it as the default signing key:
```

File: web/site/content/docs/how-to-guides/verify-signed-container-images.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+}
+EOF
+```
+
+Create a backup of the initial Kanto Container Management configuration that is found in `/etc/container-management/config.json`(the backup will be restored at the end of the guide):
+```shell
+sudo cp /etc/container-management/config.json /etc/container-management/config-backup.json
+```
+
+Configure the use of notation verifier, set it's config directory and also mark the local registry as a insecure one:
```
Comment (k-gostev) [inline web/site/content/docs/how-to-guides/verify-signed-container-images.md:87]:
```suggestion
 Configure the use of notation verifier, set its config directory, and mark the local registry as an insecure one:
```

File: web/site/content/docs/how-to-guides/verify-signed-container-images.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+```
+
+Restart the Container Management service for the changes to take effect:
+```shell
+sudo systemctl restart container-management.service
+```
+
+### Verify
+
+Create and run container from the signed image. The container prints `Hello world` to the console:
```
Comment (k-gostev) [inline web/site/content/docs/how-to-guides/verify-signed-container-images.md:112]:
```suggestion
Create and run a container from the signed image. The container prints `Hello world` to the console:
```

File: web/site/content/docs/how-to-guides/verify-signed-container-images.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+```shell
+sudo kanto-cm create --name dummy-hello --rp no --t $IMAGE
+sudo kanto-cm start --name dummy-hello --a
+```
+
+
+Make sure that docker hub hello_world image is not cached localy and verify that creating containers from it fails, as the image is not signed and the signature verification fails:
```
Comment (k-gostev) [inline web/site/content/docs/how-to-guides/verify-signed-container-images.md:119]:
```suggestion
Make sure that a docker hub hello_world image is not cached locally and verify that creating containers from it fails, as the image is not signed, and the signature verification fails: 
```

File: web/site/content/docs/how-to-guides/verify-signed-container-images.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+```shell
+sudo ctr -n kanto-cm image remove docker.io/library/hello-world:latest
+sudo kanto-cm create --name dockerhub-hello --rp no --t docker.io/library/hello-world:latest
+```
+
+### Clean up
+
+Remove the created container from Kanto Container Management and the localy cached image:
```
Comment (k-gostev) [inline web/site/content/docs/how-to-guides/verify-signed-container-images.md:127]:
```suggestion
Remove the created container from the Kanto Container Management and the locally cached image: 
```

File: web/site/content/docs/how-to-guides/verify-signed-container-images.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+```shell
```
Comment (k-gostev) [inline web/site/content/docs/how-to-guides/verify-signed-container-images.md:128]: hmm, we should add an option in the kanto-cm CLI to remove the image as well

**Review by k-gostev — CHANGES_REQUESTED** [review]

File: web/site/content/docs/how-to-guides/verify-signed-container-images.md (hunk 7 above)
Comment (dimitar-dimitrow) [reply web/site/content/docs/how-to-guides/verify-signed-container-images.md:130]: By configuring the _image_expiry_ to 0s, the locally cached images will be deleted automatically when no containers are using them.

**Review by k-gostev — APPROVED** [review]

**Pull request merged**